### PR TITLE
BasicArrayTestSuite reworked

### LINKED ISSUE #97: Refactor: Rework BasicArrayTestSuite to be more in-line with newer tests
### Description

BasicArrayTestSuite contains 32 facts at the moment, with multiple asserts per fact. This should be broken up into more facts to be more thorough.

### Script Name

BasicArrayTestSuite 

### Steps To The Task

- [x] Split up all tests within existing facts into their own facts
- [x] Add comments
- [x] Update assert descriptions

### PR DESCRIPTION
**Pull request by AdamBorek**
Facts have been broken up into multiple smaller ones

### CODE REVIEW

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 1]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
+	});
 
-	addFact("array_resize_test", function() {
+	// ARRAY RESIZE TESTS
+	
+	addFact("array_resize test #1", function() {
 
 		//#1 array_resize, array must be resized (shrink)
 		var _array = [ 1, 2, 3, 4];
 		var _newSize = 2
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
		var _newSize = 2;
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 2]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
+			{ value1: "b", value2: 8 },
+			{ value1: "ccd", value2: 7 },
+			{ value1: "cc", value2: 6 }
+		];
 			
-		//#3.2 array_sort, method based struct array sorting (strings)
+		//#6 array_sort, method based struct array sorting (strings)
 		array_sort(_array, function(_s1, _s2) {
 			return _s1.value1 > _s2.value1 ? -1 : 1;
 		})
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
		});
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 3]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
 
-	})
+	});
+	
+	// ARRAY ALL TESTS
 
-	addFact("array_all", function() {
+	addFact("array_all test #1", function() {
 	
 		var _even = function(_value, _index) { return _value % 2 == 0; }
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml:1204]:
```suggestion
		var _even = function(_value, _index) { return _value % 2 == 0; };
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 4]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
+		var _array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+		var _result = array_filter(_array, _even, -4, -4);
+		assert_array_equals(_array, [1, 2, 3, 4, 5, 6, 7, 8, 9, 10], "array_filter shouldn't mutate the input array");
+	
+	});
+	
+	// ARRAY FILTER EXT TESTS
+	
+	addFact("array_filter_ext test #", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
	addFact("array_filter_ext test #1", function() {
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 5]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
 	});
 
-	addFact("array_map", function() {
+	// ARRAY MAP TESTS
+
+	addFact("array_map test #1", function() {
 	
 		var _array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
 		
 		var _result = array_map(_array, function(_value) { return _value * 2; })
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
		var _result = array_map(_array, function(_value) { return _value * 2; });
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 6]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
 	});
 	
-	addFact("array_map_ext", function() {
+	// ARRAY MAP EXT TESTS
+
+	addFact("array_map_ext test #1", function() {
 	
 		var _array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
 		
 		var _result = array_map_ext(_array, function(_value) { return _value * 2; }, 3, 4)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
		var _result = array_map_ext(_array, function(_value) { return _value * 2; }, 3, 4);
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 7]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
+	});
+
+	addFact("array_map_ext test #2", function() {
+	
+		var _array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
 		
-		assert_equals(_result, 4, "#1 array_map_ext : fail to return the correct number of mutated elements");
-		
-		assert_array_equals(_array, [1, 2, 3, 8, 10, 12, 14, 8, 9, 10], "#2 array_map_ext : failed to correctly mutate the array");
+		var _result = array_map_ext(_array, function(_value) { return _value * 2; }, 3, 4)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
		var _result = array_map_ext(_array, function(_value) { return _value * 2; }, 3, 4);
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 8]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
-		assert_array_equals(_array, [1, 2, 3, 7, 6, 5, 4, 8, 9, 10], "#2 array_reverse_ext : failed to correctly mutate the array");
+		var _result = array_reverse_ext(_array, 3, 4);
+		assert_array_equals(_array, [1, 2, 3, 7, 6, 5, 4, 8, 9, 10], "array_reverse_ext should correctly mutate the array");
 	
 	});
 
-	addFact("array_shuffle", function() {
+	// ARRAY SHUFFLE TESTS
+	
+	addFact("array_shuffle test #", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
	addFact("array_shuffle test #1", function() {
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 9]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
 	
 		var _array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
 		
 		var _result = array_shuffle(_array);
+		assert_false(array_equals(_array, _result), "array_shuffle should shuffle the initial array");
+	
+	});
+	
+	addFact("array_shuffle test #", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
	addFact("array_shuffle test #2", function() {
```

File: projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml [hunk 10]
```diff
@@ -507,592 +783,1009 @@ function BasicArrayTestSuite() : TestSuite() constructor {
...
-		assert_false(array_equals(_array, _result), "#1 array_shuffle : failed to shuffle the initial array");
+		var _array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+		
+		var _result = array_shuffle(_array);
+		assert_array_length(_result, 10, "array_shuffle shouldn't return array element number mismatch");
 	
-		assert_array_length(_result, 10, "#2 array_shuffle : returned array element number mismatch");
+	});
+
+	addFact("array_shuffle test #", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicArrayTestSuite/BasicArrayTestSuite.gml]:
```suggestion
	addFact("array_shuffle test #3", function() {
```

**Review by Indibulous — APPROVED** [review]
Seems good! just added some missing semicolons & numbers that were missing from a few of the test names.

**Review by DiasFranciscoA — APPROVED** [review]

**Pull request merged**